States/States/Graph.c

```diff
@@ -0,0 +1,244 @@
+#include <stdlib.h>
+
+#include "Graph.h"
+
+#define INFINITY SIZE_MAX
+
+struct Graph
+{
+    size_t numberOfCities;
+    size_t** distanceBetweenCities;
+    size_t numberOfCapitals;
+    size_t** states;
+    bool* visited;
+};
+
+static size_t** createMatrix(const size_t numberOfRows, const size_t numberOfColumns, const size_t defaultValue)
+{
+    size_t** newMatrix = (size_t**)calloc(numberOfRows, sizeof(size_t*));
+    if (newMatrix == NULL)
+    {
+        return NULL;
+    }
+
+    for (size_t i = 0; i < numberOfRows; i++)
+    {
+        newMatrix[i] = (size_t*)calloc(numberOfColumns, sizeof(size_t));
+        if (newMatrix[i] == NULL)
+        {
+            for (size_t j = 0; j < i; ++j)
+            {
+                free(newMatrix[j]);
+            }
+            return NULL;
+        }
+        for (size_t j = 0; j < numberOfColumns; j++)
+        {
+            newMatrix[i][j] = defaultValue;
+        }
+    }
+    return newMatrix;
+}
+
+static void deleteMatrix(size_t*** const matrix, size_t numberOfRows)
+{
+    for (size_t i = 0; i < numberOfRows; i++)
+    {
+        free((*matrix)[i]);
+    }
+    free(*matrix);
+    *matrix = NULL;
+}
+
+static void freeMemory(size_t*** const graph, const size_t numberOfCities, size_t*** const states, const size_t numberOfCapitals, bool** const visited)
+{
+    deleteMatrix(graph, numberOfCities + 1);
+    deleteMatrix(states, numberOfCapitals);
+    free(*visited);
+    *visited = NULL;
+}
+
+void deleteGraph(Graph** graph)
+{
+    freeMemory(&(*graph)->distanceBetweenCities, (*graph)->numberOfCities, &(*graph)->states, (*graph)->numberOfCapitals, &(*graph)->visited);
+    free(*graph);
+    *graph = NULL;
+}
+
+static GraphErrorCode initCitiesAndDistanceBetweenThem(FILE* const file, Graph* graph, const size_t numberOfRoads)
+{
+    for (size_t i = 0; i < numberOfRoads; i++)
+    {
+        size_t firstCity = 0;
+        size_t secondCity = 0;
+        size_t distance = 0;
+        if (fscanf_s(file, "%Iu %Iu %Iu", &firstCity, &secondCity, &distance) != 3 || firstCity == secondCity ||
```

Comment:
```suggestion
        if (fscanf_s(file, "%lu %lu %lu", &firstCity, &secondCity, &distance) != 3 || firstCity == secondCity ||
```
В остальных местах такое же